### ISSUE TITLE
bugs in staging week starting 2018-01-13

### ISSUE THREAD

**Issue by Alys** [HabitRPG/habitica#9802]
Recently we've merged a lot of PRs to staging. Below are the bugs I've found in staging so far.

- [x] I'm in a party but on staging the website's header doesn't show my party members and has the "Battle Monsters With Friends" text as if I weren't in a party. The party page on staging does show my party. Console message: `Source map error: request failed with status 401Resource URL: https://habitrpg-staging.herokuapp.com/static/js/vendor.b9dbae6b9d38a904a42a.jsSource Map URL: vendor.b9dbae6b9d38a904a42a.js.map[Learn More]Source map error: request failed with status 401Resource URL: https://habitrpg-staging.herokuapp.com/static/js/8.0e8ca7e698d27b92e686.jsSource Map URL: 8.0e8ca7e698d27b92e686.js.map[Learn More]Source map error: request failed with status 401Resource URL: https://habitrpg-staging.herokuapp.com/static/js/app.ad1e785bb26b8fb32085.jsSource Map URL: app.ad1e785bb26b8fb32085.js.map[Learn More]Source map error: request failed with status 401Resource URL: https://habitrpg-staging.herokuapp.com/static/js/manifest.0a84c4e22970e2420ccb.jsSource Map URL: manifest.0a84c4e22970e2420ccb.js.map[Learn More]Source map error: request failed with status 401Resource URL: https://habitrpg-staging.herokuapp.com/static/js/2.c2f0d9bbce6863a6c412.jsSource Map URL: 2.c2f0d9bbce6863a6c412.js.map[Learn More]`
- [x] When you create a party in staging, you can't see it immediately. Steps to reproduce:
   1. Get a test account that has no party.
   2. Click "Invite friends" in the header.
   3. Click "Create a Party".
   4. Observe this erorr in the Console: message: "Invalid loading status \"undefined for resource at \"party\"."
   5. You can't view your new party (the portion of the page between header and footer is empty)This does not happen in production (as soon as you create a party, you can see it).
   - You can make the error go away by reloading the website but new users won't think to do that and this is a bad early experience for them.
- [x] _This happens in production too but I've only just noticed it._ In the Advanced Settings in a Challenge Habit, the "Reset "Streak" drop-down menu is enabled - it behaves as if the challenge participant is allowed to change the "Reset Streak" value (Daily, Weekly, Monthly). The participant should not be able to change it. The API correctly doesn't allow the change - when you sync you see the value chosen by the challenge creator is still in place, which is correct.
- [x] In the Advanced Settings section for a Challenge Habit, the "Adjust Streak" counter box **should** allow the challenge participant to change the streaks (for the same reasons as for Challenge Dailies; there's some points about both in the top post at https://github.com/HabitRPG/habitica/pull/9757). The form field does allow the participant to adjust the values but the participant's changes are not saved (they should be saved).
- [x] In Advanced Settings for all three task types (not Rewards): On the old website, "Stat Allocation" was visible only when you'd chosen the "Task-based" Automatic Allocation method, since it's not relevant for any other allocation method. I think I actually like that we're showing it all the time now but I think the on-hover "i" text should explain that you need to use that allocation method for the task Stats to be of any use. Otherwise users will waste time setting Stats on tasks without getting any benefit from it.
- [x] Staging > Stable > Quick Inventory > Special :  my Saddles are showing up twice. This is an urgent one.
- [x] "fetch recent messages" doesn't work in the party. You have to reload the page to see new messages.
It is working in guilds/Tavern.
- [x] In staging, when you create a challenge and want to choose the categories, the category menu appears too far up the page: http://i.imgur.com/GxRs6g1.png [EDIT: I've just noticed this happens in production too.]
- [x] When creating a Guild in staging, you can't select categories. When you try to tick the checkboxes in the Guild edit modal's category list, they don't become ticked, but if you look carefully at the My Guilds page behind the edit modal, you can see that the filter checkboxes are being ticked as you try to select categories for the new Guild. This means that it's impossible to create a guild since the categories are mandatory.

**Comment by Alys** [comment]
I've marked this as important priority and in progress because I'm assuming the staff will work on these since many must be fixed before we can deploy staging. However if that's not correct, staff should change the labels.

**Comment by Alys** [comment]
In the top post, I've ticked the bugs that are fixed. The others still happen for me. Also I've noticed this:

- [x] The delete link in the tasks's Advanced Settings should not appear for Challenge tasks.

**Comment by paglias** [comment]
Fixed the delete link for challenge tasks

**Comment by TheHollidayInn** [comment]
```
In the Advanced Settings section for a Challenge Habit, the "Adjust Streak" counter box should allow the challenge participant to change the streaks (for the same reasons as for Challenge Dailies; there's some points about both in the top post at #9757). The form field does allow the participant to adjust the values but the participant's changes are not saved (they should be saved).
```

This should be working now. I have added this to the api with a test.

**Comment by SabreCat** [comment]
* [x] Market carousel is backward: clicking the right button rotates the list rightward instead of showing you the next item rightward in the list.

<img width="996" alt="screen shot 2018-01-17 at 4 44 04 pm" src="https://user-images.githubusercontent.com/4501321/35071369-17d57f44-fba6-11e7-8757-d36c9c6541ce.png">
click >
<img width="982" alt="screen shot 2018-01-17 at 4 44 09 pm" src="https://user-images.githubusercontent.com/4501321/35071381-1e4a956c-fba6-11e7-8f4c-de2beada3c35.png">

**Comment by SabreCat** [comment]
The above is fixed, but a new carousel issue has emerged!

* [x] If you have 5 or fewer visible items in an inventory spinner, clicking left/right does not "spin" the items, but rather hides items down to 1, then cycles to showing them all again. The spinners should probably disable if they won't do anything, or at worst they could move items from the front to the back of the list or vice versa.

**Comment by SabreCat** [comment]
whoops. Tried to reference just my comment.

**Comment by paglias** [comment]
I think we can close this
